includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1586,28 +1586,71 @@ protected function get_account_statement_descriptor( $empty_value = null ) {
 	}
 
 	/**
-	 * Handles statement descriptor update when plugin settings saved.
+	 * Gets connected account business name.
+	 *
+	 * @param string $default_value Value to return when not connected or failed to fetch business name.
+	 *
+	 * @return string Business name or default value.
+	 */
+	protected function get_account_business_name( $default_value = '' ): string {
+		try {
+			if ( $this->is_connected() ) {
+				return $this->account->get_business_name();
+			}
+		} catch ( Exception $e ) {
+			Logger::error( 'Failed to get account business name.' . $e );
+		}
+
+		return $default_value;
+	}
+
+	/**
+	 * Handles connected account update when plugin settings saved.
 	 *
 	 * Adds error message to display in admin notices in case of failure.
 	 *
-	 * @param string $statement_descriptor Statement descriptor value.
+	 * @param array $account_settings Stripe account settings.
+	 * Supported: statement_descriptor, business_name, business_url, business_support_address,
+	 * business_support_email, business_support_phone, branding_logo, branding_icon,
+	 * branding_primary_color, branding_secondary_color.
 	 */
-	private function update_statement_descriptor( $statement_descriptor ) {
-		if ( empty( $statement_descriptor ) ) {
+	private function update_account( $account_settings ) {
+		if ( empty( $account_settings ) ) {
 			return;
 		}
 
-		$account_settings = [
-			'statement_descriptor' => $statement_descriptor,
-		];
-		$error_message    = $this->account->update_stripe_account( $account_settings );
+		$error_message = $this->account->update_stripe_account( $account_settings );
 
 		if ( is_string( $error_message ) ) {
-			$msg = __( 'Failed to update statement descriptor. ', 'woocommerce-payments' ) . $error_message;
+			$msg = __( 'Failed to update Stripe account. ', 'woocommerce-payments' ) . $error_message;
 			$this->add_error( $msg );
 		}
 	}
 
+	/**
+	 * Extracts and returns account-specific settings from all of plugin's settings.
+	 *
+	 * Additionally, unsets the extracted settings from all settings.
+	 *
+	 * @param array $settings Plugin settings.
+	 */
+	private function extract_account_settings( $settings ) {
+		$setting_names    = [
+			'account_statement_descriptor' => 'statement_descriptor',
+			'account_business_name'        => 'business_name',
```

Comment:
I am uneasy about this option, as it should be clear to merchants what effects change of `account_business_name` can have on their Stripe account. And I don't think we want to change an `account_business_name` as company name (on the Stripe account) and trade name (on the payment receipt) can be different.

Reply:
IMHO, keeping business name and other related settings in sync on both Stripe account and receipt makes the overall design simple and consistent. Besides, we have two types of receipts to deal with — email and print. While the print receipt will be fully managed by WCPay, Stripe will handle the email receipt. This is as per our finalized approach here pdjVC1-5Z-p2. Stripe uses data updated via its account API for generating email receipts. So, it makes sense to reuse Stripe's account data for print receipts as well to avoid duplication.

> company name (on the Stripe account) and trade name (on the payment receipt) can be different

Do we have known instances of such a scenario? If this is indeed the case, then we have a bigger problem to solve due to the limitation highlighted above. That will most likely mean freeing up Stripe from email receipt responsibility and fully managing email receipts in WCPay. This will be see a considerable bump in efforts.

Maybe we can go with a simpler design for now and introduce a separation of Stripe and receipt business names as a feature later?

Reply:
> it should be clear to merchants what effects change of account_business_name can have on their Stripe account

Great point. If we do go with my reasoning, we can then make this messaging clear on the settings page itself.